### PR TITLE
Removing set/getMisc(), docs, etc.

### PR DESCRIPTION
**Pull request by gregorgorjanc**
This is to close https://github.com/HighlanderLab/SIMplyBee/issues/584

All checkc pass on my end so I think this can be merged in, but @janaobsteter have a look please.

### CODE REVIEW

**Comment by gregorgorjanc** [comment]
I get this errors in the GitHub action

```
...
── Sitrep ──────────────────────────────────────────────────────────────────────
✖ Bootstrap 3 is deprecated; please switch to Bootstrap 5.
ℹ Learn more at
  <https://www.tidyverse.org/blog/2021/12/pkgdown-2-0-0/#bootstrap-5>.
✖ URLs not ok.
  In DESCRIPTION, URL is missing package url (http://www.simplybee.info/).
  See details in `vignette(pkgdown::metadata)`.
✔ Favicons ok.
✔ Open graph metadata ok.
✔ Articles metadata ok.
✖ Reference metadata not ok.
  In _pkgdown.yml, 3 topics missing from index: "getMisc", "mapLoci", and
  "setMisc".
  Either use `@keywords internal` to drop from index, or
...
```

but I don't see `getMisc()` anywhere in my codebase

```
grepall . getMisc
./.Rhistory
./NEWS.md
./docs/articles/A_Introduction_to_honeybee_biology.html
./docs/reference/addCastePop.html
./docs/reference/buildUp.html
./.git/logs/HEAD
./.git/logs/refs/heads/devel
./.git/COMMIT_EDITMSG
./.Rproj.user/2472D3BB/sources/session-5f57f60a/5D06C4D6-contents
./.Rproj.user/2472D3BB/sources/session-5f57f60a/7650117D-contents

grepall . setMisc
./NEWS.md
./.Rproj.user/2472D3BB/sources/session-5f57f60a/5D06C4D6-contents
./.Rproj.user/2472D3BB/sources/session-5f57f60a/7650117D-contents
```

Any clue what we have to do @janaobsteter ?

**Comment by janaobsteter** [comment]
@gregorgorjanc , this is coming from pkgdown. I know there is a list of functions that it needs so I'm guessing those docs didn't get udpated. @LStrachan was handling this, not sure if we've ever documented how to update this.

File: NEWS.md [hunk 1]
```diff
@@ -4,6 +4,28 @@ editor_options:
     wrap: 72
 ---
 
+# SIMplyBee version 0.4.2
+
+-   20??-??-??
```
Comment (janaobsteter) [inline NEWS.md:9]: @gregorgorjanc , do you want to fill this in?

**Comment by gregorgorjanc** [comment]
I have now found these autogenerated files, but indeed don't have a clue how to use the pkgdown to generate them :( Aha, we have the process described in https://github.com/HighlanderLab/SIMplyBee/issues/497, but I now have to move to another task - enough of SIMplyBee for today I am afraid. Maybe we merge this PR in @janaobsteter and deal with the pkgdown issue later in https://github.com/HighlanderLab/SIMplyBee/issues/595?

Screenshot 2024-12-16 at 08 01 16

File: R/Functions_L0_auxilary.R [hunk 2]
```diff
@@ -1280,8 +1276,6 @@ getCasteId <- function(x, caste = "all", collapse = FALSE, simParamBee = NULL) {
 #'   vector with sex information
 #' @param simParamBee \code{\link[SIMplyBee]{SimParamBee}}, global simulation parameters
 #'
-#' @seealso \code{\link[SIMplyBee]{getCaste}}
```
Comment (janaobsteter) [inline R/Functions_L0_auxilary.R:1283]: @gregorgorjanc , did you get rid of this one on purpose or by msitake?

File: vignettes/A_Honeybee_biology.Rmd [hunk 3]
```diff
@@ -340,10 +340,10 @@ offspring that are expected to be homozygous based on queen's and father's *CSD*
 alleles. You can obtain `pHomBrood` and `nHomBrood` values with the
 corresponding `pHomBrood()` and `nHombrood()` functions that can be applied
 either on the queen (`Pop` class) or colony (`Colony` class) directly. You can
-obtain the entire `misc` slot with the `getMisc()` function.
+obtain the entire `misc` slot using:
 
 ```{r misc}
-getMisc(getQueen(colony))
+getQueen(colony)@misc
```
Comment (janaobsteter) [inline vignettes/A_Honeybee_biology.Rmd:346]: Does "getMisc" not exist anymore or is this the preferred way?

**Review by janaobsteter — APPROVED** [review]
@gregorgorjanc , I've just left three tiny comments and then we can merge

File: NEWS.md (hunk 1 above)
Comment (gregorgorjanc) [reply NEWS.md:9]: Leaving empty for the release - I assume we will add more stuff later

File: R/Functions_L0_auxilary.R (hunk 2 above)
Comment (gregorgorjanc) [reply R/Functions_L0_auxilary.R:1283]: There is another one few lines below, so I removed a duplicate

File: vignettes/A_Honeybee_biology.Rmd (hunk 3 above)
Comment (gregorgorjanc) [reply vignettes/A_Honeybee_biology.Rmd:346]: Yes, no more get/setMisc(). We now just work with the list.

**Pull request merged**